src/indentation.rs

```diff
@@ -45,20 +44,61 @@
 /// ```
 /// use textwrap::indent;
 ///
-/// assert_eq!(indent(" \t  Foo   ", "->"), "-> \t  Foo   \n");
+/// assert_eq!(indent(" \t  Foo   ", "->"), "-> \t  Foo   ");
 /// ```
 pub fn indent(s: &str, prefix: &str) -> String {
     let mut result = String::new();
-    for line in s.lines() {
-        if line.chars().any(|c| !c.is_whitespace()) {
+
+    for line in CharSplitInclusive::new(s, '\n') {
```

Comment:
Could this be simplified by instead using
```rust
for idx, line in s.split('\n').enumerate() {
    // ... normal logic
    if idx > 0 {
        result.push('\n');
    }
}

I _think_ this should add exactly the name number of `\n` characters as we found originally.

Reply:
That's actually much more clever than everything I did before ^^

And it does pass the tests.